Expose the Run id to the Task instance (#1527)
Shopify/maintenance_tasks: README.md

Still, memoization can be used for throttling or reporting, and you can use
[Task callbacks](#using-task-callbacks) to persist or log a report for example.

#### Identifying the Run

A Task instance exposes the id of the Run it is being processed by as `run_id`,

> which gives you a stable identifier for a single execution of the Task:

Reviewer: Maybe add a few words to disambiguate with pauses/resumes/interruptions. I feel like the `single` here makes it ambiguous.
```suggestion
which gives you a stable identifier for the execution of the Task across interruptions and potential pauses:
```

> which gives you a stable identifier for the execution of the Task across interruptions and potential pauses:

Reviewer 2: ```suggestion
which gives you a stable identifier for the execution of the Task across
interruptions and potential pauses:
```

```ruby
class Maintenance::UpdatePostsTask < MaintenanceTasks::Task
  def collection
    Post.all
  end

  def process(post)
    post.update!(content: "Updated by run #{run_id}")
  end
end
```

It is useful to correlate log lines or reports with the Run they came from, or
as an idempotency key for work performed while processing the collection. It is
`nil` when the Task is instantiated outside of a Run, which is the case when you

> call `Maintenance::UpdatePostsTask.new` in your tests.

Reviewer 2: Nit, I'd generalize the wording here a bit, since "outside of a Run" involves understanding how we've implemented Run logic, which the typical user doesn't need to understand.
```suggestion
`nil` when a Task is instantiated outside of a typical workflow, such as when you call `Maintenance::UpdatePostsTask.new` in your tests.
```

> `nil` when a Task is instantiated outside of a typical workflow, such as when you call `Maintenance::UpdatePostsTask.new` in your tests.

Reviewer 2: Nit, let's keept he line wrapping consistent
```suggestion
`nil` when a Task is instantiated outside of a typical workflow, such as when
you call `Maintenance::UpdatePostsTask.new` in your tests.
```
